contracts/bridge-v1.yml

```diff
@@ -481,31 +601,92 @@ paths:
             application/xml:
               schema:
                 $ref: '#/components/schemas/ErrorResponse'
-  /v1/health-information/notification:
+  /v1/consent-requests/on-init:
     post:
       tags:
-        - data flow
+        - consent flow
         - hiu bridge
-      description: Notifications corresponding to events during data flow
+      summary: Response to consent request
+      description: |
+        Result of consent request creation for a patient. **id** represents the consentrequest id created by CM. The result must contain either **id** or the **error** caused. <br/>
+          Reasons for error may be
+          * Invalid references (e.g patient id, hiu id), purpose, hiTypes, ranges, persmission
       parameters:
-        - $ref: '#/components/parameters/X-CM-ID'
-        - $ref: "#/components/parameters/X-Origin-ID"
+        - $ref: "#/components/parameters/authorization"
+        - $ref: "#/components/parameters/X-HIU-ID"
       requestBody:
         required: true
         content:
           application/json:
             schema:
-              $ref: '#/components/schemas/HINotification'
+              $ref: '#/components/schemas/ConsentRequestInitResponse'
           application/xml:
             schema:
-              $ref: '#/components/schemas/HINotification'
+              $ref: '#/components/schemas/ConsentRequestInitResponse'
       responses:
-        '204':
-          description: Notification is Accepted
+        '202':
+          description: Request Accepted
         '400':
+          description:  >
+            **Causes:**
+              * Invalid data sent
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '401':
+          description:  >
+            **Causes:**
+              * Expired/Invalid token.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '500':
           description: >
             **Causes:**
-              * Invalid Request
+              * Downstream system(s) is down.
+              * Unhandled exceptions.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+  /v1/consent-requests/on-status:
```

Comment:
Is this just a Push API or it will be used when there's a pull request from HIU for enquiring the status of the consent request?

In case of PUSH (without pull) whenever CR was granted/p, there was no request to actually have

```json
{
 	"resp": {
		"requestId": ""
	}
}
```

we need to mark this field optional and make others required.

Reply:
And what is the pull URL?

Reply:
This is the callback URL whenever a HIU makes the /consent-requests/status (not implemented in reference app yet). So this is callback to the PULL url from HIU